tests/unit/test_repo.py

```diff
@@ -1,13 +1,76 @@
-import os
+from dvc.repo import Repo
+from dvc.scm.git import GitTree
+from dvc.system import System
+from dvc.utils.compat import fspath
 
-from tests.basic_env import TestDvc
 
+def test_is_dvc_internal(tmp_dir, dvc):
+    assert dvc.is_dvc_internal(fspath(tmp_dir / ".dvc" / "file"))
+    assert not dvc.is_dvc_internal("file")
 
-class TestIsDvcInternal(TestDvc):
-    def test_return_false_on_non_dvc_file(self):
-        path = os.path.join("path", "to-non-.dvc", "file")
-        self.assertFalse(self.dvc.is_dvc_internal(path))
 
-    def test_return_true_on_dvc_internal_file(self):
-        path = os.path.join("path", "to", ".dvc", "file")
-        self.assertTrue(path)
+def test_destroy(tmp_dir, dvc):
+    dvc.config.set("cache", "type", "symlink")
+
+    tmp_dir.dvc_gen("file", "text")
+    tmp_dir.dvc_gen({"dir": {"file": "lorem", "subdir/file": "ipsum"}})
+
+    dvc.destroy()
+
+    # Remove all the files related to DVC
+    assert not (tmp_dir / ".dvc").exists()
+    assert not (tmp_dir / "file.dvc").exists()
+    assert not (tmp_dir / "dir.dvc").exists()
+
+    # Leave the rest of the files
```

Comment:
could we write less assertions and cover all the cases?

Reply:
We can use a for loop :)